Ensure non-consensus node can function normally

Within the scope of this issue we need to make work normally all types of non-consensus nodes:
* account-less RPC node;
* light node;
* former consensus node.

What we need is to add enabled/disabled setting to the DBFT node settings, not to start consensus on missing account and allow consensus module to be run normally in case if authorized validator was kicked of the validators list.

Currently after #49 it's known that RPC node from single/four-nodes privnet is failing with panic, thanks to @nicolegys for pointing that out:
```
INFO [12-19|16:58:54.681] Loaded most recent local block           number=0 hash=05148f..9ce5ad td=1 age=54y9mo13h
WARN [12-19|16:58:54.681] Failed to load snapshot                  err="missing or corrupted snapshot"
INFO [12-19|16:58:54.690] Rebuilding state snapshot 
INFO [12-19|16:58:54.691] Regenerated local transaction journal    transactions=0 accounts=0
INFO [12-19|16:58:54.692] Initialized transaction indexer          limit=2,350,000
INFO [12-19|16:58:54.692] Resuming state snapshot generation       root=e38ad3..e797c5 accounts=0 slots=0 storage=0.00B dangling=0 elapsed=1.329ms
INFO [12-19|16:58:54.694] Generated state snapshot                 accounts=2 slots=0 storage=94.00B dangling=0 elapsed=3.409ms
INFO [12-19|16:58:54.699] Gasprice oracle is ignoring threshold set threshold=2
WARN [12-19|16:58:54.700] Engine API enabled                       protocol=eth
WARN [12-19|16:58:54.700] Engine API started but chain not configured for merge yet 
INFO [12-19|16:58:54.700] Starting peer-to-peer node               instance=Geth/v1.13.1-stable-1eea1e66/linux-amd64/go1.21.1
INFO [12-19|16:58:54.700] Stored checkpoint snapshot to disk       number=0 hash=05148f..9ce5ad
INFO [12-19|16:58:54.738] New local node record                    seq=1,702,994,334,737 id=4fb0a4c547ba54ca ip=127.0.0.1 udp=30307 tcp=30307
INFO [12-19|16:58:54.738] Started P2P networking                   self=enode://6d4769341e089301345c84f7ad6f1ad7721bcb9439c7ad4dc99c6789ce2c6dee57525bf1c3d21f817fcbeb8756d1b0c298586c31cc4468c8d3ded4109a4bea90@127.0.0.1:30307
INFO [12-19|16:58:54.740] IPC endpoint opened                      url=/home/anna/Documents/GitProjects/bane-labs/go-ethereum/privnet/single/node2/geth.ipc
INFO [12-19|16:58:54.740] Generated JWT secret                     path=/home/anna/Documents/GitProjects/bane-labs/go-ethereum/privnet/single/node2/geth/jwtsecret
INFO [12-19|16:58:54.741] WebSocket enabled                        url=ws://127.0.0.1:8553
INFO [12-19|16:58:54.741] HTTP server started                      endpoint=127.0.0.1:8553 auth=true prefix= cors=localhost vhosts=localhost
INFO [12-19|16:58:55.712] Unlocked account                         address=0x1dfbe2dF23Ef8B4a787dD8506ca3400B1Cd0B868
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0xe8b18a]

goroutine 4395 [running]:
github.com/ethereum/go-ethereum/consensus/dbft.(*DBFT).validatePayload(0xc0005b4900, 0xc00064b960)
	github.com/ethereum/go-ethereum/consensus/dbft/dbft.go:1274 +0x2a
github.com/ethereum/go-ethereum/consensus/dbft.(*DBFT).OnPayload(0xc0005b4900, 0xc00049ff80)
	github.com/ethereum/go-ethereum/consensus/dbft/dbft.go:1252 +0x239
github.com/ethereum/go-ethereum/eth/protocols/dbft.(*Service).HandleMessage(0xc000274540, 0xc0019f9cb0)
	github.com/ethereum/go-ethereum/eth/protocols/dbft/handler.go:105 +0x60b
github.com/ethereum/go-ethereum/eth/protocols/dbft.(*Service).Handle(0xc000274550?, 0xc0019f9cb0)
	github.com/ethereum/go-ethereum/eth/protocols/dbft/handler.go:47 +0x2d
github.com/ethereum/go-ethereum/eth/protocols/dbft.(*Service).Run(0xc000274540, 0xc00039a3f0, {0x1fda450?, 0xc000466f00?})
	github.com/ethereum/go-ethereum/eth/protocols/dbft/service.go:74 +0x172
github.com/ethereum/go-ethereum/p2p.(*Peer).startProtocols.func1()
	github.com/ethereum/go-ethereum/p2p/peer.go:426 +0x7b
created by github.com/ethereum/go-ethereum/p2p.(*Peer).startProtocols in goroutine 4384
	github.com/ethereum/go-ethereum/p2p/peer.go:424 +0x9e
```

--- a/consensus/dbft/dbft.go
+++ b/consensus/dbft/dbft.go
@@ -1113,17 +1113,15 @@ func (c *DBFT) Seal(chain consensus.ChainHeaderReader, b *types.Block, results c
 			break
 		}
 	}
-	if !isAuthorized {
-		c.lastProposalLock.RUnlock()
-		c.sealingLock.Unlock()
-		return errUnauthorizedSigner
-	}
-
-	err = c.blockQueue.SubmitTask(sealingHash, b.NumberU64(), results, stop)
-	if err != nil {
-		c.lastProposalLock.RUnlock()
-		c.sealingLock.Unlock()
-		return fmt.Errorf("failed to submit sealing task to dBFT: %w", err)
+	// Submit task to blockQueue only if we're a member of validators set, otherwise it's
+	// useless to expect that out proposal will be accepted for the current height.
+	if isAuthorized {
+		err = c.blockQueue.SubmitTask(sealingHash, b.NumberU64(), results, stop)
+		if err != nil {
+			c.lastProposalLock.RUnlock()
+			c.sealingLock.Unlock()
+			return fmt.Errorf("failed to submit sealing task to dBFT: %w", err)
+		}
 	}
 
 	c.sealingProposal = lastHeader

--- a/consensus/dbft/dbft.go
+++ b/consensus/dbft/dbft.go
@@ -1249,6 +1249,11 @@ drainLoop:
 
 // OnPayload handles Payload receive.
 func (c *DBFT) OnPayload(cp *dbftproto.Message) error {
+	if c.dbft == nil || !c.dbftStarted.Load() {
+		log.Debug("skip dBFT payload handling: dbft is inactive or not started yet", "hash", cp.Hash())
+		return nil
+	}
+
 	p := payloadFromMessage(cp)
 	// decode payload data into message
 	if err := p.decodeData(); err != nil {
@@ -1258,11 +1263,6 @@ func (c *DBFT) OnPayload(cp *dbftproto.Message) error {
 
 	if !c.validatePayload(p) {
 		log.Info("can't validate payload", "hash", cp.Hash())
-		return nil
-	}
-
-	if c.dbft == nil || !c.dbftStarted.Load() {
-		log.Info("dbft is inactive or not started yet", "hash", cp.Hash())
 		return nil
 	}
 

